wal/wal.go

```diff
@@ -170,12 +170,20 @@ func Create(lg *zap.Logger, dirpath string, metadata []byte) (*WAL, error) {
 	}
 
 	if w, err = w.renameWAL(tmpdirpath); err != nil {
-		lg.Warn(
-			"failed to rename the temporary WAL directory",
-			zap.String("tmp-dir-path", tmpdirpath),
-			zap.String("dir-path", w.dir),
-			zap.Error(err),
-		)
+		if w != nil {
+			lg.Warn(
+				"failed to rename the temporary WAL directory",
+				zap.String("tmp-dir-path", tmpdirpath),
+				zap.String("dir-path", w.dir),
+				zap.Error(err),
+			)
+		} else {
+			lg.Warn(
```

Comment:
@tedyu I agree that `w` can be `nil` in certain error cases. However, w.dir is the dirpath set on LOC 154 (`dir:      dirpath`) and it's not changed in the `renameWAL`. Instead of changing the logging, can you just create a variable (something like `logdirpath := w.dir`) right before the if loop calling renameWAL and use that instead of `w.dir` in the current code on LOC 177? That should take care of the issue and will also generate a good log message with the dir. Thanks!

Reply:
@spzala 
Your comment has been addressed.